[MRG+1] DOC Updated plot_date to NumPy/SciPy style

klaragerlei opened the pull request:
Hello,

We updated the documentation of plot_date to NumPy/SciPy style.

Thanks,
Michael, Klara

@NelleV

QuLogic commented on lib/matplotlib/axes/_axes.py:
```diff
@@ -1408,59 +1408,66 @@ def plot(self, *args, **kwargs):
                   **kwargs):
         """
-        Plot with data with dates.
+	A plot with data that contains dates.
```
These tabs need to be replaced with the correct number of spaces.

QuLogic commented:
It appears you have not [set the config for your git client](https://git-scm.com/book/en/v2/Getting-Started-First-Time-Git-Setup#Your-Identity). All your commits are attributed to "Your Name <you@example.com>", which is probably not what you want.

NelleV commented on lib/matplotlib/axes/_axes.py:
```diff
@@ -1408,59 +1408,66 @@ def plot(self, *args, **kwargs):
+	See Also
+    	--------
+    	mod : matplotlib.dates for helper functions
+        func : matplotlib.dates.date2num
```
There is a strange indentation here.
In addition, we need to revert back to the old syntax: :func:`~matplotlib.dates.date2num` for proper formatting.

NelleV commented:
Hi Michael & Klara,

The code currently has a mixture of tabs and space, due to gedit not being configured correctly. I can drop by at some point today to help you fix that.
line 1459 also appears to be too long.
Once this is fixed, I think looks great!

Thanks,
Nelle

QuLogic commented on lib/matplotlib/axes/_axes.py:
```diff
@@ -1408,59 +1408,67 @@ def plot(self, *args, **kwargs):
-        Similar to the :func:`~matplotlib.pyplot.plot` command, except
+        A plot with data that contains dates.
+        
+        Similar to the :func:~matplotlib.pyplot.plot command, except
```
The backticks should not have been removed here.

QuLogic commented on lib/matplotlib/axes/_axes.py:
```diff
@@ -1408,59 +1408,67 @@ def plot(self, *args, **kwargs):
-        Keyword arguments:
+        Note if you are using custom date tickers and formatters, it
+        may be necessary to set the formatters/locators after the call
+        to meth : plot_date since meth : plot_date will set the
```
The backticks should not have been removed here and the spaces around the colons are not necessary.

QuLogic commented on lib/matplotlib/axes/_axes.py:
```diff
@@ -1408,59 +1408,67 @@ def plot(self, *args, **kwargs):
+
+        Other Parameters
+        ----------------
+        kwargs : class : matplotlib.lines.Line2D properties : %(Line2D)s
```
`Line2D` gets expanded to several other parameters; it should be on a line of its own.

QuLogic commented:
Please remember to correct the name and committer on the first commit.

NelleV commented:
@QuLogic FYI, we did it, but it requires rebasing and some quite complex operations. I don't think that should be a requirement for contributing to matplotlib (in addition, some people may want to stay anonymous).

NelleV commented:
@klaragerlei There is still a couple of problems: I can help you with those during a break or the tutors consulting hours tonight if you'd like.

klaragerlei commented:
That would be great @Nelle, thank you!
On 7 Sep 2016 11:29, "Nelle Varoquaux" <notifications@github.com> wrote:

> @klaragerlei <https://github.com/klaragerlei> There is still a couple of
> problems: I can help you with those during a break or the tutors consulting
> hours tonight if you'd like.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/matplotlib/matplotlib/pull/7033#issuecomment-245239921>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AP4NnzXEa0B_GoXdRBinF5QGj8JmTizqks5qnpHWgaJpZM4J1KY4>
> .
>

QuLogic commented on lib/matplotlib/axes/_axes.py:
```diff
@@ -1408,59 +1408,68 @@ def plot(self, *args, **kwargs):
-                     ydate=False, **kwargs)
-
+        A plot with data that contains dates.
+        
```
There is still some whitespace on this line.

QuLogic commented:
It appears that you were able to take care of the rebasing? Just a small note that your name appears differently in some of the commits.

CI isn't happy right now, but it is raising a cryptic error. We will have to try a local build to figure out what is wrong.

QuLogic commented on lib/matplotlib/axes/_axes.py:
```diff
@@ -1408,59 +1408,68 @@ def plot(self, *args, **kwargs):
+        Other Parameters
+        ----------------
+        kwargs : class : matplotlib.lines.Line2D
+        properties : %(Line2D)s
```
`Line2D` gets expanded into several parameter listings; it should be on its own line. (And likely needs to be updated to match this formatting.)

NelleV commented:
Thanks a lot! This is looks great:

![plot_date](https://cloud.githubusercontent.com/assets/184798/18498772/f0d63360-79ed-11e6-82f6-ea662f1f1dbf.png)

QuLogic commented:
Are you okay with the some commits using your GitHub username and not your real name like the rest of them? Also, all but the second-last commit use the email of @ClaireSophie but the name is @klaragerlei. Please double-check that you will be attributed correctly.

NelleV commented:
This PR has been done as pair coding, so it is going to be hard to attribute it properly.

klaragerlei commented:
@QuLogic I think I managed to fix the double account problem, and it's all correct now, thank you for noticing.

NelleV commented:
@klaragerlei Thanks!

tacaswell commented:
backported to v2.x as 2ddc50e

QuLogic commented:
There were still two emails with the same name used for different commits; are they the same person?

klaragerlei commented:
Yes, both of them are mine, sorry, one of them is a university email
address, and the other one is personal. I realized that the university
email address might not be permanent enough, so I switched to using the
other one, but I used that email address ages ago on github for
contributing to a funny robot game, but I completely forgot (this is how
the other account appeared in my commit when I used the personal email in
the configurations), but now I deleted the old account and added both
emails to the same account, so I hope it's all fine now.

On 14 September 2016 at 22:51, Elliott Sales de Andrade <
notifications@github.com> wrote:

> There were still two emails with the same name used for different commits;
> are they the same person?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/matplotlib/matplotlib/pull/7033#issuecomment-247167062>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AP4Nn64XfFRElOwWjqnsx6NIaYRVCoAKks5qqGxcgaJpZM4J1KY4>
> .
>
